tests/functional/z_cluster/test_storagecluster_ceph_full_thresholds_params.py

```diff
@@ -0,0 +1,119 @@
+        """
+        Teardown function
+
+        """
+
+        def finalizer():
+            params_dict = {
+                "backfillFullRatio": "0.85",
+                "fullRatio": "0.8",
+                "nearFullRatio": "0.75",
+            }
+            self.set_storage_cluster_ceph_full_thresholds_params(params_dict)
+
+        request.addfinalizer(finalizer)
+
+    def test_storagecluster_ceph_full_thresholds_params(self):
+        """
+        Procedure:
+        1.Configure storagecluster CR
+        2.Wait 2 seconds
+        3.Verify ceph full thresholds parameters on cephcluster CR and storagecluster CR are same
+        4.Verify parameters with ceph api- 'ceph osd dump'
+        5.Configure the default params on storagecluster [treardown]
+
+        """
+        params_dict = {
+            "backfillFullRatio": "0.862",
+            "fullRatio": "0.793",
+            "nearFullRatio": "0.744",
+        }
+        self.set_storage_cluster_ceph_full_thresholds_params(params_dict)
+
+        logger.info("Wait 2 sec the cephcluster will updated")
+        time.sleep(2)
+
+        logger.info(
+            "Verify upgrade parameters on cephcluster CR and storagecluster CR are same"
+        )
+        cephcluster_obj = ocp.OCP(
+            kind=constants.CEPH_CLUSTER,
+            namespace=config.ENV_DATA["cluster_namespace"],
+            resource_name=constants.CEPH_CLUSTER_NAME,
+        )
+        for parameter, parameter_value in params_dict.items():
+            actual_value = cephcluster_obj.data["spec"]["storage"][parameter]
+            assert (
+                str(actual_value).lower() == str(parameter_value).lower()
+            ), f"The value of {parameter} is {actual_value} the expected value is {parameter_value}"
+
+        sample = TimeoutSampler(
+            timeout=600,
+            sleep=10,
+            func=run_cmd_verify_cli_output,
+            cmd="ceph osd dump",
+            expected_output_lst=(
+                "full_ratio 0.793",
+                "backfillfull_ratio 0.862",
+                "nearfull_ratio 0.744",
+            ),
+            cephtool_cmd=True,
+            ocs_operator_cmd=False,
+            debug_node=None,
+        )
+        if not sample.wait_for_func_status(True):
+            raise Exception(
+                "The ceph full thresholds storagecluster parameters are not updated in ceph tool"
+            )
+
+    def set_storage_cluster_ceph_full_thresholds_params(self, params_dict):
+        """
+        Configure StorageCluster CR with ceph full thresholds params
+
+        Args:
+            params_dict:
+
+        """
+        logger.info(f"Configure StorageCluster CR with upgrade params {params_dict}")
+        storagecluster_obj = ocp.OCP(
+            kind=constants.STORAGECLUSTER,
+            namespace=config.ENV_DATA["cluster_namespace"],
+            resource_name=constants.DEFAULT_CLUSTERNAME,
+        )
+        for parameter, parameter_value in params_dict.items():
```

Comment:
In case we changed the format to list, you can iterate over items in list and directly accessing values by key:

item["sc_key"]
item["value"]

If we go with this approach, we need to cahnge of argument to `sc_ceph_params (list): list of dictionaries with params to apply in SC` or something like that.

Reply:
done